foundation/mvc/models/model.php

```diff
@@ -463,6 +463,19 @@ protected function setQuery($query,array $params)
 
         $query->from([$this->modelAlias => $modelName]);
 
+        // Check if user has requested "id", if not then set it.
```

Comment:
:) Please isolate this logic in the preProcessQuery function.

What happens when rels is not set and fields are?